File: algorithms/php/1351.CountNegativeNumbersinaSortedMatrix.php
<?php

class Solution
{

    /**
     * @param Integer[][] $grid
     * @return Integer
     */
    function countNegatives($grid)
    {
        $rows = count($grid);
        $cols = count($grid[0]) - 1;

        $ans = 0;

        $count = function ($row) use ($cols) {
            $l = 0;
            $r = $cols;

            while ($l <= $r) {
                $m = $l + intval($r - $l, 2);
                if ($row[$m] >= 0) {
                    $l = $m + 1;
                } else {
                    $r = $m - 1;
                }
            }

            return $cols - $r;
        };

        foreach ($grid as $row) {
            $ans += $count($row);
        }

        return $ans;
    }
}

File: algorithms/cpp/1351.CountNegativeNumbersinaSortedMatrix.cpp
class Solution {
public:
    int count(vector<int>& row, int cols) {
        int l = 0;
        int r = cols;

        while (l <= r) {
            int m = l + (r - l) / 2;
            if (row[m] >= 0) {
                l = m + 1;
            } else {
                r = m - 1;
            }
        }

        return cols - r;
    }

    int countNegatives(vector<vector<int>>& grid) {
        int r = grid.size();
        int cols = grid[0].size() - 1;

        int ans = 0;
        for (auto row : grid) {
            ans += count(row, cols);
        }

        return ans;
    }

};